### PR TITLE
mars.d: replace C-style strings with D arrays

### PR DESCRIPTION
**Pull request by wolframw**
There is a lot of C-style string handling in mars.d (esp. `strcmp`), so I replaced most such occurrences with D arrays. One `strcmp` sitll exists (to check for lowmem), but I think converting that to a D array is not very useful.
Additionally, inside `parseCommandLine` (and in functions called from there) I have reduced calls to `toDString` where slicing from `arg` does the trick.

### CODE REVIEW

File: src/dmd/mars.d
Comment (kinke) [inline src/dmd/mars.d:249]: Lgtm after a very superficial glance, except for this here - please use `%.*s` so as not to require undocumented null-termination.

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:2424]:
```suggestion
                const(char)[] runarg = arguments[i + 1].toDString();
```

Rationale:
Space around operators, and I remember a maintainer stating a preference for functions like `toDString` having parenthesis to make it obvious they are non-trivial.

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1873]:
```suggestion
                error("unknown error style '%.*s', must be 'digitalmars' or 'gnu'", cast(int) style.length, style.ptr);
```

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1868]: Since you turned it into a switch for `-extern-std`, why not here too ?

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1802]: We have `startsWith` for this, see other examples in this module

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1674]: Same question about `switch`

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1562]: Why? Seems unrelated, unless you want to put the constraint that it is not reassign, in which case use `const` ?

File: src/dmd/mars.d
Comment (Geod24) [inline src/dmd/mars.d:1539]: Having a separate variable did make the code a bit more readable tho

**Review by Geod24 — COMMENTED** [review]
Pretty nice! Made a few style comments, but overall looks good.

File: src/dmd/mars.d
Comment (wolframw) [reply src/dmd/mars.d:1562]: I did this mainly for myself, to keep me from reassigning a slice of `arg` to `arg` itself and messing up something in another place. Probably no longer useful now that the rework is done.

File: src/dmd/mars.d
Comment (wolframw) [reply src/dmd/mars.d:1873]: It's not that easy: `error` is locally redefined in `parseCommandLine` as a function that takes 2 args: https://github.com/dlang/dmd/blob/4601b547a9fde3a03bf38cd2fa13e48670018e0d/src/dmd/mars.d#L1431
Guess I'll have to make this function variadic?

Comment (Geod24) [reply src/dmd/mars.d:1873]: Yep, although since it's simply forwarding that won't be too hard. See the other forwarding functions: https://github.com/dlang/dmd/blob/4601b547a9fde3a03bf38cd2fa13e48670018e0d/src/dmd/errors.d#L38-L51

Comment (wolframw) [reply src/dmd/mars.d:1873]: Alright, thanks.

Comment (wolframw) [reply src/dmd/mars.d]: I'm going with a variadic template now since C-style variadic function requires `extern (C)` or `extern (C++)` linkage, which then makes the `error` function unable to access the `errors` variable defined outside of the function.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**
